src/material/dialog/testing/dialog-harness.ts

```diff
@@ -63,4 +75,24 @@ export class MatDialogHarness extends ContentContainerComponentHarness<string> {
   async close(): Promise<void> {
     await (await this.host()).sendKeys(TestKey.ESCAPE);
   }
+
+  /** Gets te dialog's text. */
+  async getText() {
+    return (await this.host()).text();
+  }
+
+  /** Gets the dialog's title text. */
```

Comment:
The JsDoc for these content areas should explain that they only work if you're using the corresponding premade content element